contracts/TokenVesting.sol

```diff
@@ -2,355 +2,442 @@
-    */
-    function computeReleasableAmount(bytes32 vestingScheduleId)
-        public
-        onlyIfVestingScheduleNotRevoked(vestingScheduleId)
-        view
-        returns(uint256){
-        VestingSchedule storage vestingSchedule = vestingSchedules[vestingScheduleId];
-        return _computeReleasableAmount(vestingSchedule);
+     * @dev Computes the next vesting schedule identifier for a given holder address.
+     */
+    function computeNextVestingScheduleIdForHolder(
+        address holder
+    ) public view returns (bytes32) {
+        return
+            computeVestingScheduleIdForAddressAndIndex(
+                holder,
+                holdersVestingCount[holder]
+            );
     }
 
     /**
-    * @notice Returns the vesting schedule information for a given identifier.
-    * @return the vesting schedule structure information
-    */
-    function getVestingSchedule(bytes32 vestingScheduleId)
-        public
-        view
-        returns(VestingSchedule memory){
-        return vestingSchedules[vestingScheduleId];
+     * @dev Computes the releasable amount of tokens for a vesting schedule.
+     * @return the amount of releasable tokens
+     */
+    function _computeReleasableAmount(
+        VestingSchedule storage vestingSchedule
+    ) internal view returns (uint112) {
+        uint256 currentTime = getCurrentTime();
+        unchecked {
+            if (
+                (currentTime < vestingSchedule.cliff) || vestingSchedule.revoked
+            ) {
+                return 0;
+            } else if (
+                currentTime >= vestingSchedule.start + vestingSchedule.duration
+            ) {
+                return vestingSchedule.amountTotal - vestingSchedule.released;
+            } else {
+                // start + duration > currentTime >= cliff >= start
+                uint256 timeFromStart = currentTime - vestingSchedule.start;
+                // duration > timeFromStart >= 0
+                // uint112 * uint32 < uint256, cannot overflow
+                uint256 vestedAmount = (vestingSchedule.amountTotal *
+                    timeFromStart) / vestingSchedule.duration;
+                return uint112(vestedAmount) - vestingSchedule.released;
+            }
+        }
     }
 
     /**
-    * @dev Returns the amount of tokens that can be withdrawn by the owner.
-    * @return the amount of tokens
-    */
-    function getWithdrawableAmount()
-        public
-        view
-        returns(uint256){
-        return _token.balanceOf(address(this)).sub(vestingSchedulesTotalAmount);
+     * @notice Batch create vesting schedules.
+     * @param _vestingSchedules vesting schedules to create
+     */
+    function batchCreateVestingSchedule(
+        VestingSchedule[] calldata _vestingSchedules
+    ) external onlyOwner {
+        unchecked {
+            uint256 _vestingSchedulesTotalAmount = vestingSchedulesTotalAmount;
+            uint256 totalAmount = _vestingSchedulesTotalAmount;
```

Comment:
We don't need `_vestingSchedulesTotalAmount` here.

Use `uint256 totalAmount = vestingSchedulesTotalAmount` here. And later when comparing with previous value, simply compare `totalAmount` with `vestingSchedulesTotalAmount`.

Reply:
`vestingSchedulesTotalAmount` is in storage.